src/wallet/rpcdump.cpp

```diff
@@ -314,8 +317,11 @@ UniValue importaddress(const JSONRPCRequest& request)
     if (!request.params[2].isNull())
         fRescan = request.params[2].get_bool();
 
-    if (fRescan && pwallet->chain().getPruneMode()) {
-        throw JSONRPCError(RPC_WALLET_ERROR, "Rescan is disabled in pruned mode");
+    if (fRescan && pwallet->chain().havePruned()) {
+        // Exit early and print an error.
+        // If a block is pruned after this check, we will import the key(s),
+        // but fail the rescan with a generic error.
+        throw JSONRPCError(RPC_WALLET_ERROR, "Rescan is disabled when blocks are pruned");
```

Comment:
Would it be worth extracting this code block repeated in `importprivkey`, `importaddress`, `importpubkey`, and `importwallet` (the latter with a different error message).

```cpp
    if (fRescan && pwallet->chain().havePruned()) {
        // Exit early and print an error.
        // If a block is pruned after this check, we will import the key(s),
        // but fail the rescan with a generic error.
        throw JSONRPCError(RPC_WALLET_ERROR, "Rescan is disabled when blocks are pruned");
    }
```

Reply:
Yeah, could make sense. Will do if I have to touch for other reasons.

Reply:
faf3729242c5e9486e137e549f05bf20bd3908a0

Yeah, in a follow up is fine.